pymobiledevice3/services/mobile_image_mounter.py

```diff
@@ -233,6 +233,8 @@ def get_manifest_from_tss(self, build_manifest: Mapping) -> bytes:
                     int(tmp_build_identity['ApChipID'], 0) == chip_id:
                 build_identity = tmp_build_identity
                 break
+        else:
+            raise InternalError(f'Could not find the manifest for board {board_id} and chip {chip_id}')
```

Comment:
Let's raise `NoSuchBuildIdentityError` since technically that's the exact error.
Then, either we catch this as a general error in the project's `__main__.py`, and print this message, OR make `auto-mount` catch this error, remove the relevant files from the cache directory and then try again.